test/integration/kubectl/factory_share_transport_test.go

```diff
@@ -0,0 +1,306 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubectl
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/client-go/kubernetes"
+	"net"
+	"os"
+	"sync"
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/rest"
+	clientcmdv1 "k8s.io/client-go/tools/clientcmd/api/v1"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	kubeapiservertesting "k8s.io/kubernetes/cmd/kube-apiserver/app/testing"
+	"k8s.io/kubernetes/test/integration/framework"
+)
+
+func TestKubectlFactoryShareTransport(t *testing.T) {
```

Comment:
might also be worth exercising the clients created by the resource.Builder

that's what was tripping up repeated construction from `apply` (reproducer integration test as part of https://github.com/kubernetes/kubernetes/issues/111911 at https://github.com/liggitt/kubernetes/commits/exec-auth-reproducer demonstrating one transport, and therefore a new connection, for each item in an applied file when using exec auth)